parquet/src/file/writer.rs

```diff
@@ -390,13 +390,33 @@ impl<W: Write + Send> SerializedFileWriter<W> {
     }
 
     /// Returns a reference to the underlying writer.
+    ///
+    /// **Warning**: if you write directly to this writer, you will skip
+    /// the `TrackedWrite` buffering and byte‐counting layers. That’ll cause
+    /// the file footer’s recorded offsets and sizes to diverge from reality,
+    /// resulting in an unreadable or corrupted Parquet file.
+    ///
+    /// If you want to write safely to the underlying writer, use [`Self::write_all`].
     pub fn inner(&self) -> &W {
         self.buf.inner()
     }
 
+    /// Writes the given buf bytes to the internal buffer.
```

Comment:
It might be nice to add a note here about why you might do this

For example, something like

```suggestion
    /// Writes the given buf bytes to the internal buffer.
    ///
    /// This can be used to write raw data to an in-progress parquet file, for 
    /// example custom index structures or other payloads. Other parquet readers 
    /// will skip this data when reading the files.
```

Reply:
Thank you @alamb good suggestion.